relayer/relays/beacon/parachain-writer.go

```diff
@@ -0,0 +1,162 @@
+package beacon
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/sirupsen/logrus"
+	"github.com/snowfork/go-substrate-rpc-client/v4/types"
+	"github.com/snowfork/snowbridge/relayer/chain"
+	"github.com/snowfork/snowbridge/relayer/chain/ethereum"
+	"github.com/snowfork/snowbridge/relayer/chain/parachain"
+)
+
+type ParachainPayload struct {
+	Header   *chain.Header
+	Messages []*chain.EthereumOutboundMessage
+}
+
+type ParachainWriter struct {
+	conn        *parachain.Connection
+	payloads    <-chan ParachainPayload
+	nonce       uint32
+	pool        *parachain.ExtrinsicPool
+	genesisHash types.Hash
+}
+
+func NewParachainWriter(
+	conn *parachain.Connection,
+	payloads <-chan ParachainPayload,
+) *ParachainWriter {
+	return &ParachainWriter{
+		conn:     conn,
+		payloads: payloads,
+	}
+}
+
+func (wr *ParachainWriter) WritePayload(ctx context.Context, payload *ParachainPayload) error {
+	var calls []types.Call
+	call, err := wr.makeInitialSyncCall(payload.Header)
+	if err != nil {
+		return err
+	}
+	calls = append(calls, call)
+
+	for _, msg := range payload.Messages {
+		call, err := wr.makeMessageSubmitCall(msg)
+		if err != nil {
+			return err
+		}
+		calls = append(calls, call)
+	}
+
+	call, err = types.NewCall(wr.conn.Metadata(), "Utility.batch_all", calls)
```

Comment:
Can we remove the use of [batch_all](https://crates.parity.io/pallet_utility/pallet/struct.Pallet.html#method.batch_all) extrinsic, and just use `import_header` directly?

We used batch_all in our previous eth1 relayer to ensure that we submitted headers and event messages together for atomic processing. So that once a header was imported, the light client could verify that the submitted events were also valid.

But for eth2 I'd like us decouple header and event submission, as I think it will lead to a cleaner design.